### DISCUSSION TITLE
Check out the ESU based Leaflet I made and provide advice/critiques

### DISCUSSION THREAD

**Discussion by rory-spurr** [rory-spurr/ESAPermitsCapstone#48] [Show and tell]
Hi everyone,

In the later part of this week I have been working on getting a working Leaflet map that displays authorized take when the user selects an ESU through a dropdown menu, and a lifestage by selecting the "adult" or "juvenile" radio button. I was hoping everyone would take a look at what I have created and provide advice/critiques on how to improve it. In order to view the app, pull down the latest commits in the "main" branch of the repository, and then run the script "Making_ESU_Leaflet.R". The script will automatically run all the dependent scripts and the app should begin to run after that whole script has been run.

Alana is sick with covid which complicates the Monday meeting with the permits team. I would be fine with walking them through  what I have prepared if we still want to meet Monday, or I can prepare a series of screenshots for the permit team to look over when they have time. With me leaving for vacation on Tuesday, either of these plans work for me, and it may be worth it to wait until Alana is feeling better to meet in real time with the permits team. Let me know what you all think is best, and thanks in advance for your helpful advice!

Rory

**Comment by DianaDishman-NOAA** [comment]
Thanks Rory!

The visuals of the maps are looking great! Excited to see it come together this way - it's really taking shape (GIS pun intended).

First off, now that we can really dig into the data display I (naturally) have some questions about what we're seeing. 
> Why are we seeing negative numbers/why are they possible? If nothing is currently authorized we should just see '0' but for a few cases (to pick one example, hood Canal summer-run chum salmon adults) negative numbers pop up in the authorized take legend. 
> This reminds me that we'll want to address this with a solid QA. So after you've had a chance to look into it and you think any errors are corrected we'll pick a couple test cases, and I'll filter from my end to compare to what number is reported through the app. I'm guessing discrepancies will come down to filtering differences/errors.

I do have a next round of refinement suggestions and feedback, and I'll try to add them as issues to the task board
> Issue: please add an additional filtering radio button next to "Choose a lifestage" called "Choose an origin" to toggle between "all" or "naturally-produced".  The "all" would be a sum of the hatchery-and natural-origin fish (so all values for "production" except "unlisted hatchery" 
> Issue: please test out other color scales, or at least reverse this one. For some reason (probably related to my own biases) the lighter yellows are associated with fewer numbers and darker purples with more, even though your actual scale is the reverse. So let's try to find something more intuitive if we can, without using red/green or red/blue scales.

Also, I came up with the following question I'll send the permit team - 
> Would it be more helpful to have a set scale for Authorized Take adults and juveniles (i.e., the scale doesn't fluctuate by species)? On the one hand, having a flexible scale makes it easier to distinguish higher from lower levels among HUCs, but on the other, it might be helpful to keep the values and color scale consistent across the species so interpretation is easier.

Lastly, it's still on my list to clean up the messy HUCs I can and keep thinking about what to do for our marine/offshore areas. Putting additional thought into what to do there would be helpful as well. 

Thanks for all the great work -
Diana

**Reply by annebeau** [reply]
In case I ever had any doubt...this has proven that I am not cut out to be a computer scientist, haha. I have had a hard time running the script because I am getting errors reading in the shapefiles. I'm updating R, R Studio, and all of my packages to see if that helps. I'm going to keep plugging away and see how far I get. Will keep you posted.

**Reply by annebeau** [reply]
It appears that the .shx file is corrupt...or at least R is telling me that. I'll have to get back to troubleshooting later, as my parenting portion of the day is about to begin.

**Reply by annebeau** [reply]
Ok...progress...I forgot that you need to run the code inside GitHub desktop. I was painstakingly downloading each file into a different directory. Oops.

Here are a few initial observations / suggestions:
When you zoom out too far or click the 'reset view' button, it zooms to a panoramic tiled image of the world map:
![image](https://user-images.githubusercontent.com/103963044/183109665-415bf157-fefd-4cc7-95c9-4d0a1800f70f.png)

The + / - arrows overlap the dropdown menu so it's hard to see the species names unless so scroll so they're at the top.

As a "naive user", I would want to know what the authorized take is over what time period.

Agree with Diana's recommendations; I like the current color scale but agree with reversing it.

Nice work so far!

**Comment by rory-spurr** [comment]
Thanks for the helpful comments everyone! I believe that Alana and I have incorporated everything you all have suggested, so whenever you have time to re-run the app and check it out again feel free to leave more critiques!

Diana - I figured out the mysterious negative numbers in some of the HUCs. They actually aren't negative at all, rather the "minus" sign you are seeing is a tickmark from the legend. The reason no colors are being displayed is there are no "real" HUCs for those ESU's (i.e. they are either '99999999' or NA in value). I imagine this will be fixed when we deal with those HUC values, so hopefully we can talk more about that on Wednesday.

**Comment by DianaDishman-NOAA** [comment]
Thanks for the sleuthing Rory!  That is helpful. 

Looking at the changes, the reverse color scale looks good.  I am noticing from the scales though that some of these actual numbers are looking questionable (for example, hundreds of adults is much larger than we would allow, so these might be observe/harass?) so I need to schedule some time to do a side-by-side QA compared to hand-filtering the input file.  I get the feeling there are some inclusions/exclusions happening that I haven't specified correctly.  

The radio button for origin looks good, but there also appears to be an error with the 'all' category, which should be the sum of the natural and hatchery-origin fish, but sometimes appears blank (none) even when there are natural-origin fish to display.  Let's simplify this by 1) excluding unlisted hatchery fish, and 2) only having 'natural' and 'listed hatchery' as options for origin, so at least for now there's no additional summing that has to happen for it to display correctly. 

We can discuss further on Wednesday, but I'm also interested in the status of the lethal take, as a subset of total take? What you have represented so far is a great holistic picture of research on the landscape, but where we're truly going to push back and limit people is the total mortalities they are authorized, moreso than the number of fish they intend to handle. 

Thanks for all your work on this!
